src/slip10.rs

```diff
@@ -0,0 +1,291 @@
+                chains: vec![
+                    TestChain {
+                        chain: Chain::empty(),
+                        chain_code: "90046a93de5380a72b5e45010748567d5ea02bbf6522f979e05c0d8d8ca9fffb",
+                        private_key: "2b4be7f19ee27bbf30c667b642d5f4aa69fd169872f8fc3059c08ebae2eb19e7",
+                    },
+                    TestChain {
+                        chain: Chain::from_u32_hardened(vec![0]),
+                        chain_code: "8b59aa11380b624e81507a27fedda59fea6d0b779a778918a2fd3590e16e9c69",
+                        private_key: "68e0fe46dfb67e368c75379acec591dad19df3cde26e63b93a8e704f1dade7a3",
+                    },
+                    TestChain {
+                        chain: Chain::from_u32_hardened(vec![0, 1]),
+                        chain_code: "a320425f77d1b5c2505a6b1b27382b37368ee640e3557c315416801243552f14",
+                        private_key: "b1d0bad404bf35da785a64ca1ac54b2617211d2777696fbffaf208f746ae84f2",
+                    },
+                    TestChain {
+                        chain: Chain::from_u32_hardened(vec![0, 1, 2]),
+                        chain_code: "2e69929e00b5ab250f49c3fb1c12f252de4fed2c1db88387094a0f8c4c9ccd6c",
+                        private_key: "92a5b23c0b8a99e37d07df3fb9966917f5d06e02ddbd909c7e184371463e9fc9",
+                    },
+                    TestChain {
+                        chain: Chain::from_u32_hardened(vec![0, 1, 2, 2]),
+                        chain_code: "8f6d87f93d750e0efccda017d662a1b31a266e4a6f5993b15f5c1f07f74dd5cc",
+                        private_key: "30d1dc7e5fc04c31219ab25a27ae00b50f6fd66622f6e9c913253d6511d1e662",
+                    },
+                    TestChain {
+                        chain: Chain::from_u32_hardened(vec![0, 1, 2, 2]),
+                        chain_code: "8f6d87f93d750e0efccda017d662a1b31a266e4a6f5993b15f5c1f07f74dd5cc",
+                        private_key: "30d1dc7e5fc04c31219ab25a27ae00b50f6fd66622f6e9c913253d6511d1e662",
+                    },
+                    TestChain {
+                        chain: Chain::from_u32_hardened(vec![0, 1, 2, 2, 1000000000]),
+                        chain_code: "68789923a0cac2cd5a29172a475fe9e0fb14cd6adb5ad98a3fa70333e7afa230",
+                        private_key: "8f94d394a8e8fd6b1bc2f3f49f5c47e385281d5c17e65324b0f62483e37e8793",
+                    },
+                ],
+            },
+            // https://github.com/satoshilabs/slips/blob/master/slip-0010.md#test-vector-2-for-ed25519
+            TestVector {
+                seed: "fffcf9f6f3f0edeae7e4e1dedbd8d5d2cfccc9c6c3c0bdbab7b4b1aeaba8a5a29f9c999693908d8a8784817e7b7875726f6c696663605d5a5754514e4b484542",
+                master_chain_code: "ef70a74db9c3a5af931b5fe73ed8e1a53464133654fd55e7a66f8570b8e33c3b",
+                master_private_key: "171cb88b1b3c1db25add599712e36245d75bc65a1a5c9e18d76f9f2b1eab4012",
+                chains: vec![
+                    TestChain {
```

Comment:
The testcase for `Chain::empty()` present at https://github.com/satoshilabs/slips/blob/master/slip-0010.md#test-vector-2-for-ed25519 is missing.
This is the only test missing from the entire specification so we should definitely also add it.

Reply:
Alright, no problem. I will add this testcase.